lib/streamlit/config.py

```diff
@@ -1301,18 +1301,23 @@ def func_with_lock():
 
 
 def _validate_theme() -> None:
-    theme_opts = get_options_for_section("theme")
+    # Import logger locally to prevent circular references
+    from streamlit.logger import get_logger
+
+    LOGGER = get_logger(__name__)
 
-    reserved_theme_names = {"auto", "dark", "light"}
+    theme_opts = get_options_for_section("theme")
     theme_name = cast(str, theme_opts["name"])
-    if theme_name and theme_name.lower() in reserved_theme_names:
-        raise RuntimeError('theme.name cannot be "Auto", "Dark", or "Light".')
+
+    if theme_name and theme_name.lower() in theme.RESERVED_THEME_NAMES:
+        LOGGER.warning('theme.name cannot be "Auto", "Dark", or "Light".')
```

Comment:
I believe `name` is required so by us clearing it, we're not taking any theming into account so I'm kind of leaning towards error instead of warning?

Reply:
`name` is optional since we default to the name `Custom Theme` on the client if it's unset and a theme is defined (I double-checked that this is the case)